File: fizzbuzz.c
#include <stdio.h>
#include <stdlib.h>

void fizzbuzz(int n)
{
    int i;
    for(i = 1; i <= n; i++)
    {
        if( i % 15 == 0 )
            printf("FizzBuzz\n");
        else if( i % 3 == 0)
            printf("Fizz\n");
        else if( i % 5 == 0 )
            printf("Buzz\n");
        else
            printf("%d\n", i);
    }
    return;
}

int main(int argc, char **args)
{
    if( argc < 2)
    {
        printf("Need a number!\n");
        exit(1);
    }
    fizzbuzz(atoi(args[1]));
}

File: fizzbuzz.py
#!/usr/bin/python
import sys

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print "Need number argument > 0!"
        exit(-1)

    num = int(sys.argv[1]);
    for n in range(1,num+1):
        if n % 15 == 0:
            print "FizzBuzz"
        elif n % 5 == 0:
            print "Buzz"
        elif n % 3 == 0:
            print "Fizz"
        else:
            print "%d" % n
